Twitter Link

teodorus-nathaniel opened the pull request:
Remove anonymous login functionality, and add twitter login instead

olehmell commented on src/components/auth/ProfileModal/contents/evm-linking/LinkEvmAddressContent.tsx:
```diff
@@ -35,20 +38,35 @@ function LinkEvmAddressContent({ evmAddress, setCurrentState }: ContentProps) {
-            Unlink EVM address
-          </Button>
+          {!canUnlinkAddress ? (
+            <PopOver
+              yOffset={8}
+              trigger={<div className='w-full'>{unlinkButton}</div>}
+              triggerOnHover
+            >
+              <p>You need at least 1 identity/account linked to your account</p>
+            </PopOver>
+          ) : (
+            unlinkButton
```
This one is copy-paste in EVM and Polkadot files

teodorus-nathaniel replied:
refactored

olehmell commented on src/components/navbar/Navbar/AuthErrorModal.tsx:
```diff
@@ -0,0 +1,50 @@
+        <div className='flex flex-col rounded-2xl bg-background-lighter p-4'>
+          <span className='mb-1 text-sm text-text-muted'>Reason for ban</span>
+          <span>{blockedData?.reason.reasonText}</span>
+          {/* TODO: add href to google forms */}
```
Put this one:
https://forms.gle/ahD8j8we4L8asSVEA

olehmell commented on src/hooks/useAddressRandomSeed.ts:
```diff
@@ -0,0 +1,18 @@
+import { getLinkedIdentityQuery } from '@/services/datahub/identity/query'
+import { getAccountDataQuery } from '@/services/subsocial/evmAddresses'
+
+export default function useAddressRandomSeed(
```
Why is it Random Seed?

teodorus-nathaniel replied:
Renamed to useAddressIdentityId
so its random seed because its used as seed for the random name and random color

olehmell commented on src/pages/api/datahub/post.ts:
```diff
@@ -70,15 +71,24 @@ export type DatahubMutationBody =
       action: 'notify-update-failed'
       payload: UpdatePostBlockchainSyncStatusInput
     }
+  | {
```
Make sense to remove it if you will use separate file

teodorus-nathaniel commented on .github/workflows/feature-based.yaml:
```diff
@@ -81,6 +81,10 @@ jobs:
+            GH_TWITTER_CLIENT_ID=TEhDUEpGNEdEdlVPT1ZWdTcxbFA6MTpjaQ
+            GH_TWITTER_CLIENT_SECRET=feApUrjC8rpvkzhQUaPnRpfedabJe0uOTNWdFBv9YTMkFCMczN
+            GH_NEXTAUTH_SECRET=secret
+            GH_NEXTAUTH_URL=https://${{ env.name }}-grillchat.subsocial.network
```
yes it have to be added, I added it, just plain https://grill.chat

for the secrets, yes I have used ${{ secrets.TWITTER_CLIENT_ID }} and for others too
so need to add
secrets.TWITTER_CLIENT_ID
secrets.TWITTER_CLIENT_SECRET
secrets.NEXTAUTH_SECRET

teodorus-nathaniel commented:
> > * x_signin_started
> 
> Change to: x_login_started
> 
> > * click_name_link (I added link to name icons tooltip, this event is when user clicks on it)
> >   Change to: idenity_link_clicked, eventSource: chat (ALex wants to add all links into account modal, like on sub.id creators)
> 
> And add twitterLinked as a user prop
> 
> Btw, if loading all user data is too much for launch_app event, we can create "user_update" event and send user properties when they are ready

Updated the events, and I also optimized the launch event, so it fetches in parallel
